pkg/cri/config/config.go

```diff
@@ -398,6 +398,13 @@ func ValidatePluginConfig(ctx context.Context, c *PluginConfig) error {
 			config := c.Registry.Configs[endpoint]
 			config.Auth = &auth
 			c.Registry.Configs[endpoint] = config
+			// set both endpoint and host for tls/auth configs
+			if endpoint != u.Host {
```

Comment:
I think the variable name chooses on line 388 is maybe confusing... those should all be of the form host[:port] the filtering of scheme on line 396 is a nice helper..

I'm not sure what the case is where endpoint (better named hostAndOptionalyPort) is not = u.Host?